waspc/packages/wasp-config/__tests__/testFixtures.ts

```diff
@@ -0,0 +1,486 @@
+/**
+ * This module contains sample data that can be used for testing purposes.
+ * In our case the sample data represents UserSpec data.
+ */
+
+import { GET_USER_SPEC } from '../src/_private.js'
+import * as UserApi from '../src/userApi.js'
+
+/**
+ * This type removes all properties from T that are optional.
+ */
+type MinimalConfig<T> = {
+  [K in keyof T as undefined extends T[K] ? never : K]: T[K]
+}
+
+/**
+ * Required alias for domain consistency sake.
+ */
+type FullConfig<T> = Required<T>
+
+type ConfigType = 'minimal' | 'full'
+
+export function createUserApp(configType: ConfigType): UserApi.App {
+  if (configType === 'minimal') {
+    return createMinimalUserApp()
+  } else if (configType === 'full') {
+    return createFullUserApp()
+  } else {
+    throw new Error(`Unknown config type: ${configType}`)
+  }
+}
+
+export function createUserSpec(configType: ConfigType): UserApi.UserSpec {
```

Comment:
Why not call `createUserApp` here, to not repeat the logic? Maybe you don't even need this function then, just do `createUserApp(configType)[GET_USER_SPEC]()`? And if the purpose was to hide they ugly part of it, define helper `getUserSpec()` and then `createUserSpec(createUserApp(configType))`.